test/OpenTelemetry.Tests/Resources/ResourceTest.cs

```diff
@@ -158,7 +158,6 @@ public void CreateResource_NotSupportedAttributeTypes()
                 { "dynamic", new { } },
                 { "array", new int[1] },
                 { "complex", this },
-                { "float", 0.1f },
```

Comment:
should we add int and float for the supported tests?

Reply:
I can do this, I think we can make it a separate but similar [Fact] method since this one is specifically about types we don't support?

Reply:
well, we will support, right? Just to prevent a code duplication I would add in the `Supported` method and add a comment that that isn't supported by default, but we will do a `nice` thing and help

Reply:
Sorry, just clarifying your vision for this test/tests:
CreateResource_**NotSupportedAttributeTypes**() {
--test for dynamic object--
--test for complex object (this)--
}

CreateResource_**SupportedAttributeTypes**(){
--test for string--
--test for long--
--test for bool--
--test for double--
}

I think we agree on the above inclusions for sure. We are now converting ints and floats that are passed in, we add these in the **SupportedAttributeTypes** method with maybe an inline comment above stating this disclaimer?

The other thing is we don't currently support arrays of primitives but the spec prescribes it, I'll currently leave the test for arrays in **NotSupportedAttributeTypes**() ?

Thank you!

Reply:
for the `int` and `float`, i would add in the `CreateResource_SupportedAttributeTypes` and in the validation, where you would have to check for `1L` or `1.0d`, I would add a comment explaining why the test for int/float is mapping to long/double.

For the array, I would wait for cijo/codeblanch to answer...